api/internal/target/kusttarget_test.go

```diff
@@ -313,19 +313,21 @@ configurations:
 	th.WriteF("/merge-config/name-prefix-rules.yaml", `
 namePrefix:
 - path: metadata/name
-  apiVersion: v1
+  group: apps
+  version: v1
   kind: Deployment
 - path: metadata/name
-  apiVersion: v1
+  version: v1
   kind: Secret
 `)
 	th.WriteF("/merge-config/name-suffix-rules.yaml", `
 nameSuffix:
 - path: metadata/name
-  apiVersion: v1
```

Comment:
I don't know why we use `apiVersion` here.
Because we don't allow the use of the `apiVersion` field, we must use the `group` and `version` fields.
https://github.com/kubernetes-sigs/kustomize/blob/37715863f06accc8494fb52f30d69dc1827ceb09/api/types/fieldspec.go#L30

Reply:
This is strange indeed. Perhaps a residue from a previous implementation?